### PR TITLE
[dataloader] added a train_as_val data for evaluation

### PR DESCRIPTION
**Pull request by sngImagry**
data from the train set which is processed as validation set

### CODE REVIEW

File: soundbay/train.py [hunk 1]
```diff
@@ -95,6 +104,8 @@ def modeling(
     model = models_dict[model_args.pop('_target_')](**model_args)
 
 
+    print('*** model has been loaded successfully ***')
+    print(f'number of trainable params: {sum([p.numel() for p in model.parameters() if p.requires_grad]):,}')
```
Comment (mosheman5) [inline soundbay/train.py:108]: like!

File: soundbay/trainers.py [hunk 2]
```diff
@@ -70,7 +72,8 @@ def train(self):
             self.logger.reset_losses()
             self.train_epoch(epoch)
             self.epochs_trained += 1
-            self.eval_epoch(epoch)
+            self.eval_epoch(epoch, 'val')
+            self.eval_epoch(epoch, 'train_as_val')
```
Comment (mosheman5) [inline soundbay/trainers.py:76]: perhaps we could add flag for `eval_train_as_val_interval` so we won't have to go over the whole train set every epoch, because it might be much larger than the validation set. I think a default of once per 10 epoch could suffice, wdyt?

**Review by mosheman5 — CHANGES_REQUESTED** [review]
Looks great! Great feature for improving our model and training analysis.
I suggest adding a parameters for testing the train only once in x epochs so we won't lose much time training the models.

**Review by mosheman5 — APPROVED** [review]
